lib/db/mem.js

```diff
@@ -251,14 +294,28 @@ module.exports = function (log, error) {
 
   Memory.prototype.accountDevices = function (uid) {
     return getAccountByUid(uid)
-      .then(function(account) {
-        var devices = Object.keys(account.devices).map(
-          function (id) {
-            return account.devices[id]
-          }
-        )
-        return P.resolve(devices)
-      })
+      .then(
+        function(account) {
+          return Object.keys(account.devices).map(
+            function (id) {
+              var device = devices[internalDeviceId(account, { id: id })]
+              var session = sessionTokens[device.sessionTokenId]
+              if (session) {
+                device.uaBrowser = session.uaBrowser
+                device.uaBrowserVersion = session.uaBrowserVersion
+                device.uaOS = session.uaOS
+                device.uaOSVersion = session.uaOSVersion
+                device.uaDeviceType = session.uaDeviceType
+                device.lastAccessTime = session.lastAccessTime
+              }
+              return device
+            }
+          )
+        },
+        function (err) {
+          return []
+        }
+      )
```

Comment:
Clearly this changes the semantics of `accountDevices` from "list all the session tokens" to "list all the devices", which makes sense.

Seeing the UA information joined on here like this, I wonder if we should just create a device record for *all* sessions, put this information in that record directly, and remove it from the sessionToken table.  At the least, we should ensure every login with `service=sync` receives a device record, even if it's from a client too old to register itself explicitly.  Otherwise weird stuff will happen with e.g. the journeybuilder events.

If not, I think we should keep an API for listing all the sessionTokens on the account, we're going to need to do that sooner or later regardless of whether they qualify as devices or not.